### ISSUE TITLE
ConstantExpectedAnalyzer CA1857 does not trigger on constructor calls

### ISSUE THREAD

**Issue by jafaber** [dotnet/roslyn-analyzers#7831]
Decorating a constructor with `[ConstantExpected]` does nothing. The analyzer basically ignores it.

Minimal code example that shows the problem:

```cs
internal class C
{
    private C([ConstantExpected(Min = 3, Max = 4)] int value) { }
    private static void M([ConstantExpected(Min = 3, Max = 4)] int value) { }

    private static void A()
    {
        M(0); // Correctly triggers: CA1857: The constant does not fit within the value bounds of '3' to '4'
        var obj = new C(0); // BUG: Does NOT trigger CA1857
    }
}
```

At least in the documentation I've seen, this does not seem to be deliberate.

AI is saying that the below Invocation registration (as found in current analyzer source code) does not cover constructor calls. For that it suggests adding a similar registration for `OperationKind.ObjectCreation`. I have not (yet) investigated this any further.

```cs
context.RegisterOperationAction(context => OnInvocation(context, constantExpectedContext), OperationKind.Invocation);
```

**Comment by Frulfump** [comment]
Just an FYI. From the repo README.md


>Repository Consolidation
The Microsoft.CodeAnalysis.NetAnalyzers package moved into the dotnet/sdk repository for further development and respond to issues formerly in this repository.
>
>All other analyzers moved into the dotnet/roslyn repository.

**Comment by jafaber** [comment]
@Frulfump  Thanks. Turned out to be in dotnet/sdk as well.

Created here now: https://github.com/dotnet/sdk/issues/54740
